File: src/resources.py
'''Contains resources and Pygame objects needed for FLUKE RPG.'''

import pygame

pygame.init()
pygame.mixer.init()
pygame.display.set_mode()

# Terrain images.
BEACH = pygame.image.load("art/beach.png").convert_alpha()
WATER = pygame.image.load("art/water.png").convert_alpha()
HOUSE = pygame.image.load("art/house.png").convert_alpha()
TREE = pygame.image.load("art/tree.png").convert_alpha()

# Decoration images.
DECORATIONS = {
    "attack": pygame.image.load("art/cat3.png").convert_alpha(),
    "corpse": pygame.image.load("art/corpse.png").convert_alpha()
    }

# Player images.
MAX_LEVEL = 5
PLAYER = {}
for i in range(MAX_LEVEL):
    PLAYER[i+1] = pygame.image.load(
        "art/fighter{}.png".format(i+1)
        ).convert_alpha()

# Enemy images.
ENEMY_IMAGES = {
    "steven": pygame.image.load("art/male-head.png"),
    "sierra": pygame.image.load("art/female-head.png"),
    "rabbit": pygame.image.load("art/rabbitside.png")
    }

# Sound cache.
SOUNDTRACK = pygame.mixer.Sound("music/digging-for-riches.ogg")
SFX_HIT = pygame.mixer.Sound("music/sfx/hit.ogg")
SFX_LEVELUP = pygame.mixer.Sound("music/sfx/levelup.ogg")

# Fonts.
SMALL_FONT = pygame.font.Font(None, 14)
LARGE_FONT = pygame.font.Font(None, 20)

File: js/app/resources.js
// Contains page objects needed for BioQuest
define(['domReady!'], function(document) {
  "use strict";

  var res = {};

  function loader() {
    function load_image(id) {
      return document.getElementById(id);
    }

    res = {
      image: {
        attack: load_image("im-attack"),
        beach: load_image("im-beach"),
        grass: load_image("im-grass"),
        corpse: load_image("im-corpse"),
        house: load_image("im-house"),
        water: load_image("im-water"),
        tree: load_image("im-tree"),
        flame: load_image("im-flame"),
        "burning-tree": load_image("im-burning-tree"),
        "burning-house": load_image("im-burning-house")
      },
      creature: {
        fighter: {
          to: 8
        },
        steven: {
          to: 2
        },
        sierra: {
          to: 3
        },
        rabbit: {
          to: 1
        },
        cat: {
          to: 3
        }
      },
      portraits: {
        "Narrator": load_image("im-narrator"),
        "Wesley": load_image("im-wesley-scott"),
        "Anna": load_image("im-anna-parker")
      },
      param: {
        protagonist_level: document.getElementById("param-protagonist-level"),
        protagonist_xp: document.getElementById("param-protagonist-xp"),
      },
      active: {
        save: document.getElementById("active-save"),
        load: document.getElementById("active-load")
      }
    };

    // Load player images.
    for (var creature in res.creature) {
      for (var i = 1; i <= res.creature[creature].to; i++) {
        res.creature[creature][i] = document.getElementById(
          "ip-" + creature + "-" + i
          );
      }
    }

    // Button commands.
    var Active = {
      save() {
        requirejs('game').Game.save();
      },
      load() {
        load_game();
      }
    };

    // Activate buttons.
    for (var acv in res.active) {
      res.active[acv].addEventListener('click', Active[acv]);
    }
  }

  loader();
  return res;
});
